src/components/layer.d.ts

```diff
@@ -1,15 +1,27 @@
+import * as MapboxGL from "mapbox-gl";
+import {PureComponent} from "react";
+
+export interface LayerProps {
+  id?: string;
+  type:  'fill' | 'line' | 'symbol' | 'circle' | 'fill-extrusion' | 'raster' | 'background' | 'heatmap' | 'hillshade';
+  source?: string;
+  beforeId?: string;
+  layout?: MapboxGL.AnyLayout;
+  paint:
+    | MapboxGL.BackgroundPaint
+    | MapboxGL.FillPaint
+    | MapboxGL.FillExtrusionPaint
+    | MapboxGL.LinePaint
+    | MapboxGL.SymbolPaint
+    | MapboxGL.RasterPaint
+    | MapboxGL.CirclePaint
+    | MapboxGL.HeatmapPaint
+    | MapboxGL.HillshadePaint;
+  filter?: any[];
+  minzoom?: number;
+  maxzoom?: number;
+}
+
+export default class Layer extends PureComponent<LayerProps> {}
```

Comment:
Does it matter that this does not match the implementation (functional component)?

Reply:
I did this so it would have the ref prop, I don't think it truly matters but alternatively we could define the ref prop to accomplish the same thing

Reply:
I kinda like that approach more so I switched it